src/schema/v2/infiniteDiscovery/discoverArtworks.ts

```diff
@@ -179,55 +179,96 @@ export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
       user.seenArtworks?.map((node) => node.internalID) || []
     const artworkIdsToFilter = [...likedArtworkIds, ...seenArtworkIds]
 
-    let searchQuery
-
+    let response
     if (likedArtworkIds.length > 0) {
-      searchQuery = gql`
-      {
-        Get {
-          InfiniteDiscoveryArtworks(
-              nearObject: {
-                beacon: "${beacon}",
-                certainty: ${certainty}
-              },
-              limit: ${limit},
-              offset: ${offset},
-          ) {
-            internalID
-            _additional {
-              id
+      const searchQuery = gql`
+        {
+          Get {
+            InfiniteDiscoveryArtworks(
+                nearObject: {
+                  beacon: "${beacon}",
+                  certainty: ${certainty}
+                },
+                limit: ${limit},
+                offset: ${offset},
+            ) {
+              internalID
+              _additional {
+                id
+              }
             }
           }
         }
-      }
-    `
-    } else {
-      searchQuery = gql`
-      {
-        Get {
-          InfiniteDiscoveryArtworks(
-              limit: ${limit},
-              offset: ${offset},
-          ) {
-            internalID
-            _additional {
-              id
+      `
+
+      const challengeQuery = gql`
+        {
+          Get {
+            InfiniteDiscoveryArtworks(
+              limit: 200
+              where: {
+                path: ["isCurated"]
+                operator: Equal
+                valueBoolean: true
+              }
+            ) {
+              internalID
+              _additional {
+                id
+              }
             }
           }
         }
-      }`
-    }
+      `
 
-    const body = await weaviateGraphqlLoader({
-      query: searchQuery,
-    })()
+      const nearObjectResponse = await weaviateGraphqlLoader({
+        query: searchQuery,
+      })()
+      const challengeResponse = await weaviateGraphqlLoader({
+        query: challengeQuery,
+      })()
 
-    if (!body.data.Get.InfiniteDiscoveryArtworks)
-      return connectionFromArray([], args)
+      response = nearObjectResponse?.data?.Get?.InfiniteDiscoveryArtworks
 
-    const artworkIds = body.data.Get.InfiniteDiscoveryArtworks.map(
-      (node) => node.internalID
-    )
+      // Insert the challenge artwork at a random index
+      sampleSize(
+        challengeResponse.data?.Get?.InfiniteDiscoveryArtworks,
+        2
+      ).forEach((challengeArtwork) => {
+        const length = response.length
+        const randomIndex = random(length)
+        // Splice mutates the array
+        response.splice(randomIndex, 0, challengeArtwork)
```

Comment:
**suggestion**: we shouldn't mutate the response to avoid any potential side effects (due to many factors like for example schema change for example or similar)

```suggestion
      // consider renaming this to curated
      const challengeArtworks = sampleSize(
        challengeResponse.data?.Get?.InfiniteDiscoveryArtworks,
        2
      );
      // Combine response and challenge artworks, then shuffle
      const shuffledArtworks = shuffle(response.concat(challengeArtworks));
```

Reply:
>we shouldn't mutate the response to avoid any potential side effects 

I don't follow. In this instance we're just mutating an array of "internalIDs", which shouldn't be affected by schema changes? But will update to copy the array instead of mutating directly. 

>const shuffledArtworks = shuffle(response.concat(challengeArtworks));

This doesn't quite work since order matters in this instance. The `nearObject` search is returning works in order of most recommended to less, so we need to preserve that order.